validation.rst

```diff
@@ -303,6 +303,14 @@ In Symfony, constraints are similar: they are assertions that a condition
 is true. Given a value, a constraint will tell you if that value
 adheres to the rules of the constraint.
 
+Most constraints are only validated when the value is non-empty, i.e., it's not
```

Comment:
nice addition imo but the paragraph is a bit long, maybe a rephrase is needed, even though I can not propose a better one

maybe adding it in an info/notice box could also be great

Reply:
In the reference for the individual constraints we have this little sentence:

> As with most of the other constraints, null and empty strings are considered valid values. This is to allow them to be optional values. If the value is mandatory, a common solution is to combine this constraint with NotBlank.

I think we could take that and reword it in a general manner to be reused here.